### ISSUE TITLE
Docker CLI commands hanging after upgrade to 4.34.0

### ISSUE THREAD

**Issue by gmeans** [docker/for-mac#7411]
### Description

After upgrading to v4.34.0 I've noticed the docker CLI "hangs" in certain situations requiring a full restart to get things working again. I've noticed this primarily when using VS Code dev containers as when in this state, they won't start up.

When in this state, the CLI won't return until the command is cancelled, then returning the desired results successfully without error as seen in the screenshot.

<img width="637" alt="image" src="https://github.com/user-attachments/assets/1ab46f44-8cb3-41af-bc6e-5f941c20b89a">

__

### Reproduce

1. It's necessary to be in whatever state is causing this. A fresh start of Docker normally clears this.
1. Run `docker version --format '{{json . }}'` 
1. It will hang, hit ctrl+c.
1. The output returns.

### Expected behavior

The intended behavior is for the command to return immediately without having to cancel it.

### docker version

```bash
Client:
 Version:           27.2.0
 API version:       1.47
 Go version:        go1.21.13
 Git commit:        3ab4256
 Built:             Tue Aug 27 14:14:45 2024
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.34.0 (165256)
 Engine:
  Version:          27.2.0
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.21.13
  Git commit:       3ab5c7d
  Built:            Tue Aug 27 14:15:41 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.7.20
  GitCommit:        8fc6bcff51318944179630522a095cc9dbf9f353
 runc:
  Version:          1.1.13
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    27.2.0
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.2-desktop.1
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.2-desktop.2
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.34
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-debug
  desktop: Docker Desktop commands (Alpha) (Docker Inc.)
    Version:  v0.0.15
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-desktop
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.25
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.3.0
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.13.0
    Path:     /Users/REDACTED/.docker/cli-plugins/docker-scout

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 6
 Server Version: 27.2.0
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fc6bcff51318944179630522a095cc9dbf9f353
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.10.4-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 8
 Total Memory: 7.655GiB
 Name: docker-desktop
 ID: 634ca12e-19cd-4dcf-a4e2-fb115d22c81c
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=unix:///Users/REDACTED/Library/Containers/com.docker.docker/Data/docker-cli.sock
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

3BA6725A-0109-40B3-814F-CCE4FAE48533/20240905150721

### Additional Info

Please note that the diagnostics and info commands did not run successfully until I restarted Docker Desktop and cleared containers.

**Comment by xeron** [comment]
Happening for me too. Seems like docker doesn't wake up from resource saver mode.

**Comment by bmhatfield** [comment]
Also happening for me. It seems strongly coupled to resource saver mode.

**Comment by jasosims** [comment]
Same here. Running some other command, such as `docker compose pull`, wakes it up, and then I can start my containers as usual.

**Comment by djs55** [comment]
Thanks for the report. There's a bug where some commands (e.g. `docker ps --filter` and `docker compose up`) will not wake the VM properly. Running a pull or another container usually does the trick. One workaround would be to disable Resource Saver in settings. Alternatively you could try a nightly build which has the fix, the latest is:

- [Apple Silicon Docker.dmg](https://desktop-stage.docker.com/mac/main/arm64/166283/Docker.dmg)
- [Intel Docker.dmg](https://desktop-stage.docker.com/mac/main/amd64/166283/Docker.dmg)

**Comment by mxfactorial** [comment]
issue resolved with [Intel Docker.dmg](https://desktop-stage.docker.com/mac/main/amd64/166283/Docker.dmg)

**Comment by medyagh** [comment]
I am curious is there any timeline when the fix would be released for public ? at minikube we are contemplating to do a patch release, but if docker desktop fix is on the way we wouldnt have to do that

**Comment by djs55** [comment]
@medyagh we discussed it today and decided to create a DD patch release (4.34.2), aiming for this Thursday (12th). Hopefully this avoids the need for Minikube to do a patch release.

**Comment by bmhatfield** [comment]
Thank you for fixing this! It works for me now!

**Comment by elvismon** [comment]
Echoing the good news: `minikube start` works again using the docker driver. Thank you!

**Comment by djs55** [comment]
Thanks for the report and for letting us know it's fixed!

**Comment by MihaelaStoica** [comment]
Closing this ticket, as the fix has been included in Docker Desktop 4.34.2. See [release notes](https://docs.docker.com/desktop/release-notes/#4342)
